src/Components/WalletModal/index.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect } from 'react';
+import { Modal } from 'minimal-react-modal';
+import { useWalletModalOpen, useWalletModalToggle } from '../../contexts/Application';
+import { SUPPORTED_WALLETS } from "../../constants";
+import WalletOption from './WalletOption';
+import IF from "../IF";
+import {useWeb3React, usePrevious} from "../../hooks";
+import PrimaryButton from '../PrimaryButton';
+
+const WALLET_VIEWS = {
+    OPTIONS: 'options',
+    OPTIONS_SECONDARY: 'options_secondary',
+    ACCOUNT: 'account',
+    PENDING: 'pending'
+}
+
+const WalletModal = props => {
+
+    const { active, account, activate, error, connector } = useWeb3React();
+    const walletModalOpen = useWalletModalOpen();
+    const toggleWalletModal = useWalletModalToggle();
+    const [pendingWallet, setPendingWallet] = useState();
+    const [pendingError, setPendingError] = useState();
+    const [errorMessage, setErrorMessage] = useState("");
+    const [walletView, setWalletView] = useState(WALLET_VIEWS.OPTIONS);
+
+    const tryActivation = async connector => {
+        setPendingWallet(connector) // set wallet for pending view
+        setWalletView(WALLET_VIEWS.PENDING)
+        activate(connector, undefined, true).catch(e => {
+            setPendingError(true);
+            setErrorMessage(e.message);
+        })
+    }
+
+    // close modal when a connection is successful
+    const activePrevious = usePrevious(active)
+    const connectorPrevious = usePrevious(connector)
+    useEffect(() => {
+        if (walletModalOpen && ((active && !activePrevious) || (connector && connector !== connectorPrevious && !error))) {
```

Comment:
maybe let's format this a bit better

Reply:
https://github.com/Alexintosh/PIE-landing/pull/30/commits/7fb47dbb2650ae4701e804eaef777dfd750e65f9 This should make it a bit more readable